optimum/intel/openvino/modeling_base.py

```diff
@@ -366,6 +366,9 @@ def _from_transformers(
 
         save_dir = TemporaryDirectory()
         save_dir_path = Path(save_dir.name)
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
```

Comment:
do you think this could be moved to `OVBaseModel.__init__()`  as done in : https://github.com/huggingface/optimum-intel/blob/2c79d987b685ef5e87b2062518598e9c1cf64c78/optimum/intel/openvino/modeling_diffusion.py#L106-115

Reply:
That would require changing some more logic; it doesn't work out of the box. Do you want me to do that?

In https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_base.py#L368 the TemporaryDirectory is converted to a Path, which is then past to both `main_export` and `_from_pretrained`. So `OVBaseModel__init__()` `model_save_dir` is not a `TemporaryDirectory` but a `Path`. And to prevent the TemporaryDirectory from automatically being deleted, `_model_save_dir_tempdirectory_instance` must be set to the actual TemporaryDirectory instance.

Reply:
thanks for the update, for now I think we can merge and will see if it can be added in a following PR